pkg/volmount/volmount_test.go

```diff
@@ -0,0 +1,245 @@
+package volmount
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+	"testing"
+
+	"github.com/Microsoft/go-winio/vhd"
+	"github.com/hpe-storage/common-host-libs/windows/wmi"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+// fullObjectIDFromObjectID turns the given class and objectID into an objectID suitable for calling
+// wmi.ExecWmiMethod on.
+// There should be an __PATH property on the object for this, according to wbemtest.exe, but I cannot see
+// how to get it from the objects exposed by wmi.
+func fullObjectIDFromObjectID(class, objectID string) string {
+	return class + `.ObjectId="` + strings.ReplaceAll(strings.ReplaceAll(objectID, `\`, `\\`), `"`, `\"`) + `"`
+}
+
+// turnRawDiskIntoAVolume will take the given PhysicalDisk path, **wipe its partition table** and
+// create a single new partition, returning the Volume path needed by the mount-point APIs.
+func turnRawDiskIntoAVolume(physPath string) (string, error) {
```

Comment:
I wasn't sure if that API would actually work for a random VHDX (as opposed to a Differencing VHDX for use as a WCOW writable layer), and also, I wasn't sure if a circular dependency via test-suite was desirable, since hcsshim is one place I plan to use these functions.

Happy to give it a burl though.

I did find some a different (non-WMI-based) set of APIs to do this, but I would have had to wrap those up for Go as well, as they involved lots of unions of `X_GPT` and `X_MBR`.

Reply:
I had a quick look, and I'm still left with about half this function doing WMI calls even after calling `FormatWritableLayerVhd`, as I need to hunt down the `\\Volume{GUID}` for the new volume (which as far as I know means enumerating all volumes). The WMI code received the volume-path as a side-effect of the partitioning, but `FormatWritableLayerVhd` doesn't seem to provide any such output, since it hands all that off to HCS.

Reply:
Yeah looks like as of now we have no better way of doing all this. Let's keep the WMI code for now.

Reply:
Actually I just remembered something. You might be able to get the `\\Volume{GUID}` path by using another HCS API from here: https://github.com/microsoft/hcsshim/blob/master/computestorage/mount.go#L14. I am not exactly sure if it returns the same path that you want but if it works it would be great.

Reply:
Hmm, interesting. I believe using computestorage makes RS5 the minimum supported version though. I still worry about the circular dependency though.

Reply:
That worked, and is much shorter and neater.

Minor nit, `computestorage.GetLayerVhdMountPath` doesn't include the final `\` that all these Win32 APIs expect/return.

A more-annoying change, `computestorage.FormatWritableLayerVhd` loses the original error (it stringifies with `fmt.ErrorF` instead of propagating with `fmt.WrapF`, so the check for "run as non-Admin" goes from
```
if errno, ok := errors.Cause(err).(syscall.Errno); ok && errno == syscall.ERROR_PRIVILEGE_NOT_HELD
```
to
```
strings.Contains(err.Error(), "A required privilege is not held by the client.")
```
(and I hope that's not a localised string...)

Reply:
This last "more-annoying change" could be reverted after https://github.com/microsoft/hcsshim/pull/925 lands.

Reply:
And that's landed, so if this has not been merged by the time I next look at this, I'll pull the latest hcsshim (only used by the tests) and return this to the `errors.Cause`-based check.